### DISCUSSION TITLE
Tanner Linsley - TanStack

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#145] [Announcements]
# Background 

- https://twitter.com/tannerlinsley
- https://twitter.com/NozzleIO
- https://tanstack.com

### Nozzle and Open Source

- [x] What is nozzle? 
- [x] Why put out so much of your code as open source code? 
- [x] how do you manage running a company and also maintaining so many popular tools?
- [x] Initially you seemed very hesitant to TypeScript but now your libraries are shining examples of great TypeScript APIs. How did you opinion change on TS and do you feel it's an essential part of your workflow now?

### Query

- [ ] What is react query? What problem were you trying to solve?
- [ ] In react it's generally bad to "fetch as you render", does query help you solve this?
- [ ] It initially started out as a react only library (and many of your other libs) but now support multiple front end frameworks. What prompted you to do that (since it's pretty unique) and has it mad developing those libraries harder?
- [ ] Query has devltools, what do they do? Do they integrate with other tech in the tanstack?
- [ ] What are some of you favorite features in query that you love using?

### Router

- [x] Why another router? What does this one do differently?
- [ ] It it a completely client side router? or does it do server rendering too?
- [x] How does keeping your state in the URL make apps simpler? Does it have pitfalls?
- [x] What magic makes all of the routing typesafe?!
- [ ] How does it help you with code splitting?
- [ ] This has preloading too, what are you preloading and how can that enhance an application?
- [x] How do loaders and actions fit in? to me they seem like query split into separate modules for the router
- [x] Do you plan to bring this library to other frameworks too?

### Future

- [ ] What is tanstack start?
- [ ] What technologies does it bring together? (What's bling)
- [ ] Where do you want to take this?

**Comment by hipstersmoothie** [comment]
https://github.com/total-typescript/ts-reset
https://github.com/studio-freight/lenis/

**Comment by tannerlinsley** [comment]
https://observablehq.com/collection/@observablehq/plot

**Comment by just-be-dev** [comment]
- https://web.hypothes.is/
- https://pragdave.me/discover/unison/2023-03-11-abilities/
